### PR TITLE
chore(common): CHECKOUT-6713 Switch tslint to nx eslint

### PR DESCRIPTION
**Pull request by animesh1987**
## What?
Switch tslint for nx lint mechanism

## Why?
As part of migrating to Nx, we should be able to use all the benefits provided like enforcing module boundaries and the lint configuration that is provided by Nx out of the box. Also caching of previous lint runs.

Fixes the issue of imports from other modules.

## Testing / Proof
- Build
- Test

@bigcommerce/checkout

### CODE REVIEW

File: packages/core/src/payment/strategies/apple-pay/apple-pay-payment.mock.ts [hunk 1]
```diff
@@ -33,5 +34,5 @@ export class MockApplePaySession {
         return event;
     }
 
-    oncancel() { }
+    oncancel() { noop }
```
Comment (davidchin) [inline packages/core/src/payment/strategies/apple-pay/apple-pay-payment.mock.ts:37]: Not sure if this is correct.

File: packages/core/src/payment/strategies/bolt/bolt.mock.ts [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+export function getBoltClientScriptMock(shouldSucceed = false, isValidTransactionReference = true): BoltCheckout {
     return {
+        /* eslint-disable */
         configure: jest.fn((_cart: object, _hints: {}, callbacks?: BoltCallbacks) => {
-            return getConfiguredBoltMock(shouldSucceed, isValidTransactionReference, callbacks || { success: () => {}, close: () => {}});
+            /* eslint-enable */
+            return getConfiguredBoltMock(
+                shouldSucceed,
+                isValidTransactionReference,
+                callbacks || { success: () => noop, close: () => noop });
```
Comment (davidchin) [inline packages/core/src/payment/strategies/bolt/bolt.mock.ts:12]:
```suggestion
                callbacks || { success: noop, close: noop });
```

File: packages/core/src/payment/strategies/braintree/braintree-payment-processor.spec.ts [hunk 3]
```diff
@@ -488,7 +489,7 @@ describe('BraintreePaymentProcessor', () => {
                     .mockImplementation(({ addFrame }) => {
                         addFrame();
 
-                        return new Promise(() => { });
+                        return new Promise(() => noop);
```
Comment (davidchin) [inline packages/core/src/payment/strategies/braintree/braintree-payment-processor.spec.ts:492]:
```suggestion
                        return new Promise(noop);
```

File: packages/core/src/payment/strategies/braintree/braintree-payment-processor.spec.ts [hunk 4]
```diff
@@ -498,7 +499,7 @@ describe('BraintreePaymentProcessor', () => {
             it('cancels card verification', async () => {
                 braintreePaymentProcessor
                     .challenge3DSVerification('tokenization_nonce', 122)
-                    .catch(() => {});
+                    .catch(() => noop);
```
Comment (davidchin) [inline packages/core/src/payment/strategies/braintree/braintree-payment-processor.spec.ts:502]:
```suggestion
                    .catch(noop);
```

File: packages/core/src/payment/strategies/braintree/braintree-visacheckout-payment-strategy.ts [hunk 5]
```diff
@@ -55,8 +56,8 @@ export default class BraintreeVisaCheckoutPaymentStrategy implements PaymentStra
                 }
 
                 const {
-                    onError = () => {},
-                    onPaymentSelect = () => {},
+                    onError = () => noop,
```
Comment (davidchin) [inline packages/core/src/payment/strategies/braintree/braintree-visacheckout-payment-strategy.ts:59]:
```suggestion
                    onError = noop,
```

File: packages/core/src/payment/strategies/braintree/braintree-visacheckout-payment-strategy.ts [hunk 6]
```diff
@@ -55,8 +56,8 @@ export default class BraintreeVisaCheckoutPaymentStrategy implements PaymentStra
...
+                    onPaymentSelect = () => noop,
```
Comment (davidchin) [inline packages/core/src/payment/strategies/braintree/braintree-visacheckout-payment-strategy.ts:60]:
```suggestion
                    onPaymentSelect = noop,
```

File: packages/core/src/payment/strategies/chasepay/chasepay-payment-strategy.spec.ts [hunk 7]
```diff
@@ -119,7 +120,7 @@ describe('ChasePayPaymentStrategy', () => {
 
         jest.spyOn(paymentStrategyActionCreator, 'widgetInteraction');
         jest.spyOn(requestSender, 'post');
-        JPMC.ChasePay.showLoadingAnimation = jest.fn(() => jest.fn(() => {}))
+        JPMC.ChasePay.showLoadingAnimation = jest.fn(() => jest.fn(() => noop))
```
Comment (davidchin) [inline packages/core/src/payment/strategies/chasepay/chasepay-payment-strategy.spec.ts:123]:
```suggestion
        JPMC.ChasePay.showLoadingAnimation = jest.fn(() => jest.fn(noop))
```

File: packages/core/src/payment/strategies/checkoutcom-custom/checkoutcom-sepa/checkoutcom-fawry-payment-strategy.spec.ts [hunk 8]
```diff
@@ -67,7 +67,7 @@ describe('CheckoutcomFawryPaymentStrategy', () => {
             .mockReturnValue(getPaymentMethod());
 
         jest.spyOn(formPoster, 'postForm')
-            .mockImplementation((_url, _data, callback = () => {}) => callback());
+            .mockImplementation((_url, _data, callback = () => noop) => callback());
```
Comment (davidchin) [inline packages/core/src/payment/strategies/checkoutcom-custom/checkoutcom-sepa/checkoutcom-fawry-payment-strategy.spec.ts:70]:
```suggestion
            .mockImplementation((_url, _data, callback = noop) => callback());
```

File: packages/core/src/payment/strategies/checkoutcom-custom/checkoutcom-sepa/checkoutcom-sepa-payment-strategy.spec.ts [hunk 9]
```diff
@@ -74,7 +74,7 @@ describe('CheckoutcomSEPAPaymentStrategy', () => {
             .mockReturnValue(getPaymentMethod());
 
         jest.spyOn(formPoster, 'postForm')
-            .mockImplementation((_url, _data, callback = () => {}) => callback());
+            .mockImplementation((_url, _data, callback = () => noop) => callback());
```
Comment (davidchin) [inline packages/core/src/payment/strategies/checkoutcom-custom/checkoutcom-sepa/checkoutcom-sepa-payment-strategy.spec.ts:77]:
```suggestion
            .mockImplementation((_url, _data, callback = noop) => callback());
```

File: packages/core/src/payment/strategies/clearpay/clearpay-payment-strategy.spec.ts [hunk 10]
```diff
@@ -49,8 +49,8 @@ describe('ClearpayPaymentStrategy', () => {
     let strategy: ClearpayPaymentStrategy;
 
     const clearpaySdk = {
-        initialize: () => {},
-        redirect: () => {},
+        initialize: () => noop,
```
Comment (davidchin) [inline packages/core/src/payment/strategies/clearpay/clearpay-payment-strategy.spec.ts:52]:
```suggestion
        initialize: noop,
```

File: packages/core/src/payment/strategies/clearpay/clearpay-payment-strategy.spec.ts [hunk 11]
```diff
@@ -49,8 +49,8 @@ describe('ClearpayPaymentStrategy', () => {
...
+        redirect: () => noop,
```
Comment (davidchin) [inline packages/core/src/payment/strategies/clearpay/clearpay-payment-strategy.spec.ts:53]:
```suggestion
        redirect: noop,
```

File: packages/core/src/payment/strategies/clearpay/clearpay-payment-strategy.spec.ts [hunk 12]
```diff
@@ -131,10 +131,10 @@ describe('ClearpayPaymentStrategy', () => {
             .mockReturnValue(Promise.resolve(clearpaySdk));
 
         jest.spyOn(clearpaySdk, 'initialize')
-            .mockImplementation(() => {});
+            .mockImplementation(() => noop);
```
Comment (davidchin) [inline packages/core/src/payment/strategies/clearpay/clearpay-payment-strategy.spec.ts:134]:
```suggestion
            .mockImplementation(noop);
```

File: packages/core/src/payment/strategies/clearpay/clearpay-payment-strategy.spec.ts [hunk 13]
```diff
@@ -131,10 +131,10 @@ describe('ClearpayPaymentStrategy', () => {
...
 
         jest.spyOn(clearpaySdk, 'redirect')
-            .mockImplementation(() => {});
+            .mockImplementation(() => noop);
```
Comment (davidchin) [inline packages/core/src/payment/strategies/clearpay/clearpay-payment-strategy.spec.ts:137]:
```suggestion
            .mockImplementation(noop);
```

File: packages/core/src/payment/strategies/clearpay/clearpay-payment-strategy.ts [hunk 14]
```diff
@@ -73,7 +74,7 @@ export default class ClearpayPaymentStrategy implements PaymentStrategy {
         await this._redirectToClearpay(countryCode, state.paymentMethods.getPaymentMethod(methodId, gatewayId));
 
         // Clearpay will handle the rest of the flow so return a promise that doesn't really resolve
-        return new Promise(() => {});
+        return new Promise(() => noop);
```
Comment (davidchin) [inline packages/core/src/payment/strategies/clearpay/clearpay-payment-strategy.ts:77]:
```suggestion
        return new Promise(noop);
```

**Review by davidchin — COMMENTED** [review]
Could you take a look at the use of `noop`? I haven't commented every single one of them.

File: packages/core/src/payment/strategies/apple-pay/apple-pay-payment.mock.ts (hunk 1 above)
Comment (animesh1987) [reply packages/core/src/payment/strategies/apple-pay/apple-pay-payment.mock.ts:37]: Changed it to do a return instead `oncancel() { return }`

**Review by davidchin — APPROVED** [review]
Nice work!

**Pull request merged**